src/DynamoCore/Nodes/NodeCategories.cs

```diff
@@ -163,6 +166,88 @@ public static string PreprocessTypeName(string fullyQualifiedName)
         }
 
         /// <summary>
+        /// This method returns a name for the icon based on name of the node.
+        /// </summary>
+        /// <param name="descriptor">Function descriptor, that contains all info about node.</param>
+        /// <param name="overridePrefix">
+        /// overridePrefix is used as default value for generating node icon name.
+        /// If overridePrefix is empty, it uses QualifiedName property.
+        /// e.g. Autodesk.DesignScript.Geometry.CoordinateSystem.ByOrigin
+        /// </param>
+        public static string TypedParametersToString(FunctionDescriptor descriptor, string overridePrefix = "")
+        {
+            var builder = new StringBuilder();
+
+            foreach (TypedParameter tp in descriptor.Parameters)
+            {
+                string typeOfParameter = tp.Type.ToString();
+
+                // Check if there simbols like "[]".
+                // And remove them, according how much we found.
+                // e.g. bool[][] -> bool2
```

Comment:
This can be rephrased as:

```csharp
// Check to see if there is array indexer symbols '[]', if so turn their 
// dimensionality into a number (e.g. 'bool[][]' turned into 'bool2').
```